eclair-core/src/main/scala/fr/acinq/eclair/router/SyncHandlers.scala

```diff
@@ -0,0 +1,203 @@
+
+  def handleSendChannelQuery(d: Data, s: SendChannelQuery)(implicit ctx: ActorContext, log: LoggingAdapter): Data = {
+    implicit val sender: ActorRef = ctx.self // necessary to preserve origin when sending messages to other actors
+    // ask for everything
+    // we currently send only one query_channel_range message per peer, when we just (re)connected to it, so we don't
+    // have to worry about sending a new query_channel_range when another query is still in progress
+    val query = QueryChannelRange(s.chainHash, firstBlockNum = 0L, numberOfBlocks = Int.MaxValue.toLong, TlvStream(s.flags_opt.toList))
+    log.info("sending query_channel_range={}", query)
+    s.to ! query
+
+    // we also set a pass-all filter for now (we can update it later) for the future gossip messages, by setting
+    // the first_timestamp field to the current date/time and timestamp_range to the maximum value
+    // NB: we can't just set firstTimestamp to 0, because in that case peer would send us all past messages matching
+    // that (i.e. the whole routing table)
+    val filter = GossipTimestampFilter(s.chainHash, firstTimestamp = Platform.currentTime.milliseconds.toSeconds, timestampRange = Int.MaxValue)
+    s.to ! filter
+
+    // clean our sync state for this peer: we receive a SendChannelQuery just when we connect/reconnect to a peer and
+    // will start a new complete sync process
+    d.copy(sync = d.sync - s.remoteNodeId)
+  }
+
+  def handleQueryChannelRange(channels: SortedMap[ShortChannelId, PublicChannel], routerConf: RouterConf, origin: RemoteGossip, q: QueryChannelRange)(implicit ctx: ActorContext, log: LoggingAdapter): Unit = {
+    implicit val sender: ActorRef = ctx.self // necessary to preserve origin when sending messages to other actors
+    ctx.sender ! TransportHandler.ReadAck(q)
+    Kamon.runWithContextEntry(remoteNodeIdKey, origin.nodeId.toString) {
+      Kamon.runWithSpan(Kamon.spanBuilder("query-channel-range").start(), finishSpan = true) {
+        log.info("received query_channel_range with firstBlockNum={} numberOfBlocks={} extendedQueryFlags_opt={}", q.firstBlockNum, q.numberOfBlocks, q.tlvStream)
+        // keep channel ids that are in [firstBlockNum, firstBlockNum + numberOfBlocks]
+        val shortChannelIds: SortedSet[ShortChannelId] = channels.keySet.filter(keep(q.firstBlockNum, q.numberOfBlocks, _))
+        log.info("replying with {} items for range=({}, {})", shortChannelIds.size, q.firstBlockNum, q.numberOfBlocks)
+        val chunks = Kamon.runWithSpan(Kamon.spanBuilder("split-channel-ids").start(), finishSpan = true) {
+          split(shortChannelIds, q.firstBlockNum, q.numberOfBlocks, routerConf.channelRangeChunkSize)
+        }
+
+        Kamon.runWithSpan(Kamon.spanBuilder("compute-timestamps-checksums").start(), finishSpan = true) {
+          chunks.foreach { chunk =>
+            val reply = Router.buildReplyChannelRange(chunk, q.chainHash, routerConf.encodingType, q.queryFlags_opt, channels)
```

Comment:
buildReplyChannelRange should be in this file too.

Reply:
Done with dc79e8d.